### PR TITLE
fix logging

### PR DESCRIPTION
**Pull request by SaranshBaniyal**
-replace timestamp with shorter utc timestamp #13 
-add STDOUT loader #14 
-clean up Query log (was logged twice)
-display the txnHash when executed correctly
-add colored prefix for library logs
-add a flag to enable/disable library's logging

### CODE REVIEW

File: packages/wasmkit/src/lib/deploy/contract.ts [hunk 1]
```diff
@@ -45,6 +45,25 @@ export class Contract {
   private checkpointData: Checkpoints;
   private readonly checkpointPath: string;
 
+  private loadInterval?: NodeJS.Timer;
+
+  private printLoadingAnimation (logMsg: string): void {
+    const loadingChars = ['/', '-', '\\', '|'];
```
Comment (udit-gulati) [inline packages/wasmkit/src/lib/deploy/contract.ts:52]: fix the typo of `\\`

File: packages/wasmkit/src/lib/deploy/contract.ts [hunk 2]
```diff
@@ -45,6 +45,25 @@ export class Contract {
...
+    let index = 0;
+
+    this.loadInterval = setInterval(() => {
+      process.stdout.write(`${logMsg} ${loadingChars[index]}\r`);
```
Comment (udit-gulati) [inline packages/wasmkit/src/lib/deploy/contract.ts:56]: Why can't we use console.log() here?

Also, logMsg shoudl be after loading char

File: packages/wasmkit/src/lib/deploy/contract.ts [hunk 3]
```diff
@@ -203,7 +222,7 @@ export class Contract {
       this.env.runtimeArgs.command === "test"
         ? `deploy ${this.contractName} ${initTimestamp}`
         : label;
-    console.log(`Instantiating with label: ${label}`);
+    this.printLoadingAnimation(`Instantiating with lable: ${label}`);
```
Comment (udit-gulati) [inline packages/wasmkit/src/lib/deploy/contract.ts:226]: fix the typo

File: packages/wasmkit/src/lib/deploy/contract.ts [hunk 4]
```diff
@@ -304,6 +305,10 @@ export class Contract {
       transferAmount,
       customFees,
       memo
-    ).finally(() => { this.stopLoadingAnimation(); });
+    ).then((result) => {
+      this.stopLoadingAnimation();
+      console.log(`TransactionHash: '${chalk.green(result.transactionHash)}'`);
```
Comment (udit-gulati) [inline packages/wasmkit/src/lib/deploy/contract.ts:310]: This has to handle the fail/success result of transaction result. We only get txnHash if the result was successful and in that case returned response code is `0`, so here you can check the code value, if 0 then print sucess with txnHash, else failure with failure reason and returned code.

You can see what the structure of response is in both cases by running them manually by sending a correct and incorrect txn and logging the full response and see where the code value is in the response.

Also need to handle the color coding of success/failure with chalk

**Review by udit-gulati — CHANGES_REQUESTED** [review]

File: packages/wasmkit/src/lib/deploy/contract.ts (hunk 1 above)
Comment (SaranshBaniyal) [reply packages/wasmkit/src/lib/deploy/contract.ts:52]: Sir I have used \\ as an escape sequence to print a single backslash character

File: packages/wasmkit/src/lib/deploy/contract.ts (hunk 4 above)
Comment (SaranshBaniyal) [reply packages/wasmkit/src/lib/deploy/contract.ts]: Sir, I am having trouble locating the response code which is supposed to be checked

File: packages/wasmkit/src/lib/deploy/contract.ts [hunk 5]
```diff
@@ -264,30 +264,36 @@ export class Contract {
     return instantiateInfo;
   }
 
-  async queryMsg (msgData: Record<string, unknown>): Promise<any> { // eslint-disable-line  @typescript-eslint/no-explicit-any
+  async queryMsg (msgData: Record<string, unknown>, skipLogs?: boolean): Promise<any> { // eslint-disable-line  @typescript-eslint/no-explicit-any
```
Comment (udit-gulati) [inline packages/wasmkit/src/lib/deploy/contract.ts:267]: Set a default value here as false, or handle if only this thing is true, then print the logs

File: packages/wasmkit/src/lib/deploy/contract.ts [hunk 6]
```diff
@@ -264,30 +264,36 @@ export class Contract {
...
     if (this.contractAddress === "mock_address") {
       throw new WasmkitError(ERRORS.GENERAL.CONTRACT_NOT_INSTANTIATED, {
         param: this.contractName
       });
     }
     // Query the contract
-    this.printLoadingAnimation(`[${chalk.gray("wasmkit")}] ${chalk.green("INF")} Querying ${this.contractAddress} => ${Object.keys(msgData)[0]}`);
-
+    if (!skipLogs) {
```
Comment (udit-gulati) [inline packages/wasmkit/src/lib/deploy/contract.ts:274]: This should be if `skipLogs === true` then print the logs

**Review by udit-gulati — CHANGES_REQUESTED** [review]

File: packages/wasmkit/src/lib/deploy/contract.ts (hunk 6 above)
Comment (SaranshBaniyal) [reply packages/wasmkit/src/lib/deploy/contract.ts:274]: But sir if we want logging to be enabled by default, then I believe it should be ```if(skipLogs===false)``` as by default skipLogs is now initialized as ```skipLogs: boolean = false```. (As when skipLogs is true, we will be skipping logs, and when false we will be printing logs)

Comment (udit-gulati) [reply packages/wasmkit/src/lib/deploy/contract.ts:274]: Yeah, so just set the skipLogs param as true by default

Comment (SaranshBaniyal) [reply packages/wasmkit/src/lib/deploy/contract.ts:274]: ok sir

Comment (SaranshBaniyal) [reply packages/wasmkit/src/lib/deploy/contract.ts:274]: I have renamed the skipLogs param to logFlag as it seemed more appropriate(to minimize confusion)

**Review by udit-gulati — APPROVED** [review]

**Pull request merged**